packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1959,6 +1959,12 @@ def add_termination(
         cls.background_round_cls = background_round_cls
         cls.termination_transition_function = termination_abci_app.transition_function
         cls.termination_event = termination_event
+        new_cross_period_persisted_keys = copy(cls.cross_period_persisted_keys)
+        new_cross_period_persisted_keys.extend(
+            termination_abci_app.cross_period_persisted_keys
+        )
+        new_cross_period_persisted_keys = list(set(new_cross_period_persisted_keys))
+        cls.cross_period_persisted_keys = new_cross_period_persisted_keys
```

Comment:
This was missing leading to flakiness in the e2e tests of register_termination_abci